hashicorp/dev-portal
# fix: nested tabs responsive issue

## ✅ "Ready for Review" checklist


---

## 🔗 Relevant links

- [Preview link][preview] 🔎
- [Asana task](https://app.asana.com/0/1202097197789424/1202172254720223/f) 🎟️

## 🗒️ What

Fixes an issue where nested tabs would not detect overflow correctly.

## 🤷 Why

Nested not-default-active tabs were overflowing on small viewports.

## 🛠️ How

Updates `use-overflow-ref` to detect and observe the closest ancestor `[role="tabpanel"]` element (if any), such that when the visibility of said panel changes, overflow is re-calculated.

## 🧪 Testing

- [ ] Visit [the preview link][preview], and compare the new behaviour to [upstream](https://developer.hashicorp.com/waypoint/tutorials/get-started-docker/get-started-install)
    - Use a very small initial viewport (eg `320 px`)
    - Change first tab element from `Manual Installation` to `Linux`
    - Upstream, you should see the overflow issue when selecting the `Linux` tab
        - Specifically, the nested tabs for Linux not be in "dropdown mode", and will be overflowing off the edge of the screen
        - Resizing the viewport should trigger overflow detection, switching the nested tabs into "dropdown mode"
    - In the PR preview, the overflow issue should be resolved
        - Specifically, the nested tabs for Linux should be in "dropdown mode" on first render

## 💭 Anything else?

Not at the moment!

[preview]: https://dev-portal-git-zsaddress-tabs-responsive-issue-hashicorp.vercel.app/waypoint/tutorials/get-started-docker/get-started-install
